docs/rfc/active/azkaban-flow-job/README.md

```diff
@@ -0,0 +1,95 @@
+
+The design includes the metadata needed to represent Azkaban jobs and flows as data job entities and their relationships to other
+entities like Datasets.
+
+## Motivation
+
+Azkaban is a popular open source workflow manager created and extensively used at LinkedIn. Azkaban metadata is a critical piece
+in the metadata graph since data processing jobs are the primary driver of data movement and creation.
+
+Without job metadata, it is not possible to understand the data flow across an organization. Additionally, jobs are needed in the
+lineage graph to surface operational metadata and have a complete view of data movement and processing
+
+## Requirements
+
+The following requirements exists as part of this rfc:
+
+- Define Data flow and job as entities and model metadata for azkaban data job and flows
+- Enable Search & Discovery for Data jobs and flows
+- Link DataJob entities to existing entities like Datasets to build a more complete metadata graph
+- Automatically derive dataset upstream lineage from data job metadata (inputs and outputs)
+
+## Non Requirements
+
+Azkaban has its own application to surface jobs, flows, operational metadata and job logs. DataHub doesn't intend to be
+a replacement for it. Users will still need to go to Azkaban UI to look at logs and debug issues. DataHub will only show
+important and high level metadata in the context of search, discovery and exploration including lineage and will link to
+Azkaban UI for further debugging or finer grained information.
+
+## Detailed design
+
+![high level design](graph.png)
+
+The graph diagram above shows the relationships and high level metadata associated with Data Job and Flow entities.
+
+An Azkaban flow is a DAG of one or more Azkaban jobs. Usually, most data processing jobs consume one or more inputs and 
+produce one of more outputs (represented by datasets in the diagram). There can be other kinds of housekeeping jobs as well
+like cleanup jobs which don't have any data processing involved.
+
+In the diagram above, the Azkaban job node consumes datasets `ds1` and `ds2` and produces `ds3`. It is also linked to the
+flow it is part of. As shown in the diagram, dataset upstream lineage is derived from the azkaban job metadata which results
+in `ds1` and `ds2` being upstreams of `ds3`.
+
+### Entities
+There will be 2 top level GMA [entities](../../../what/entity.md) in the design: Data job and flow.
+
+### URN Representation
+We'll define two [URNs](../../../what/urn.md): `DataJobUrn` and `DataFlowUrn`.
+These URNs should allow for unique identification for an Data job and flow respectively.
+
+An example Data flow URN will look like below:
+```
+urn:li:dataFlow:(azkaban,flow_id,fabric)
+```
+An example Data job URN will look like below:
+```
+urn:li:dataJob:(azkaban,job_id,fabric)
+```
+
+### Azkaban Flow metadata
+
+Below is a list of metadata which can be associated with an azkaban flow:
+
+- Azkaban cluster
```

Comment:
I assume the flow ID is not unique only within the cluster? If so not including the cluster name as part of the URN would lead to clashes? Same concern for job ID.

Reply:
Updated urn definitions to include cluster.

Reply:
Can drop this from here since it's moved to part of URN already?

Reply:
Another idea: since cluster may not be a first-class thing in every scheduler system, maybe just prepend it to `flow_id`? i.e. `urn:li:dataFlow:(azkaban,cluster.flow_id)`?

Reply:
Will drop this for now. Schedulers without cluster support can always use a reasonable sentinel value for cluster, e.g. `default`.